File: patterns/c-patterns/pattern109.c
// Pattern 109: ...

// *     *     *
// **   ***   **
// *** ***** ***
// *************
// *************
// *************
// *************

#include <stdio.h>

int main()
{
    int n = 7;
    int i, j;

    int x = n;
    for (i = 1; i <= n; i++)
    {
        for (j = 1; j < 2 * n; j++)
        {
            if ((j >= x && j <= 2 * n - x) || j <= n - x + 1 || j >= n + x - 1)
            {
                printf("*");
            }
            else
            {
                printf(" ");
            }
        }
        x--;
        printf("\n");
    }
    return 0;
}

File: patterns/javascript-patterns/pattern109.js
// Pattern 109: ...

// *     *     *
// **   ***   **
// *** ***** ***
// *************
// *************
// *************
// *************

let n = 7;

let x = n;
for (let i = 1; i <= n; i++) {
    let row = "";
    for (let j = 1; j < 2 * n; j++) {
        if ((j >= x && j <= 2 * n - x) || j <= n - x + 1 || j >= n + x - 1) {
            row += "*";
        } else {
            row += " ";
        }
    }
    x--;
    console.log(row);
}
